packages/container/libraries/strongly-typed/README.md

```diff
@@ -0,0 +1,59 @@
+[![Test coverage](https://codecov.io/gh/inversify/monorepo/branch/main/graph/badge.svg?flag=%40inversifyjs%2Fstrongly-typed)](https://codecov.io/gh/inversify/monorepo/branch/main/graph/badge.svg?flag=%40inversifyjs%2Fstrongly-typed)
```

Comment:
I'm missing a section talking about container hierarchies.

Since a child container overrides a parent container, so does their binding maps:

```ts
interface ParentBindingMap {
  foo: Foo;
  bar: Bar;
}

interface ChildBindingMap extends ParentBindingMap {
  bar: AlternativeBar;
  baz: Baz;
}
```

I think it would be great to include this detail in the docs.

Reply:
Good point! I've now also strengthened the type signature of `createChild()` to enforce this, too. I've made it so that the child's binding map must extend the parent's binding map. Is this conceptually correct? ie in your example, is `AlternativeBar` meant to be compatible with `Bar`? (ie could they both be abstracted to implement some `CommonBar`)?

Reply:
`AlternativeBar` can be anything, related or unrelated to `Bar`. Maybe my example was sort of misleading :sweat_smile:

Reply:
Oh okay in that case my typings are too strong. Will adjust.

Reply:
Have actually readjusted this to automatically generate the correct binding map from a partial child map, because I think that's a friendlier API. Let me know if you disagree?

Have updated the readme, but copying here for discussion if needed:


> If the child is strongly-typed, its binding will automatically be adjusted to include the parent types:

> ```ts
> const parent = new TypedContainer<{foo: Foo}>();
> const child = parent.createChild<{bar: Bar}>();
> child.get('bar') // ok
> child.get('foo') // ok
> ```

> A child may also override a parent's bindings with a completely unrelated type:

> ```ts
> const parent = new TypedContainer<{foo: Foo}>();
> const child = parent.createChild<{foo: Bar}>();
> const resolved: Bar = child.get('foo');
> ```

Reply:
Love it